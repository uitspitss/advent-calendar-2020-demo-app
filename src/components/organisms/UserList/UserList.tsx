import React, { FC } from 'react';
import tw, { styled } from 'twin.macro';

import { User } from 'types/user';

export type UserListProps = {
  /**
   * class for twin.macro
   */
  className?: string;
  /**
   * no children
   */
  children?: never;
  /**
   * user data
   */
  users: User[];
};

type Props = UserListProps;

const Component: FC<Props> = (props) => {
  const { className, users } = props;

  return (
    <div className={className}>
      {users.map(({ id, name }) => (
        <p key={name}>
          {id}
          {': '}
          {name}
        </p>
      ))}
    </div>
  );
};

const StyledComponent = styled(Component)``;

/**
 * TODO: write an explanation of UserList
 */
export const UserList: FC<UserListProps> = (props) => (
  <StyledComponent {...props} />
);
