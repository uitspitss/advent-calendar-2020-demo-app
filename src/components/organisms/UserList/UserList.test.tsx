import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { UserList } from './UserList';

describe('organisms/UserList', () => {
  const users = [
    { id: 1, name: 'john' },
    { id: 2, name: 'lisa' },
  ];

  it('Snap Shot', () => {
    const tree = renderer.create(<UserList users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Behavior', () => {
    beforeEach(() => {
      render(<UserList users={users} />);
    });

    test.todo('something of test');
  });
});
