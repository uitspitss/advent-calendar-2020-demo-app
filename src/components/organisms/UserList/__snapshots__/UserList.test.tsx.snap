// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/UserList Snap Shot 1`] = `
<div
  className="css-ghlsvi-StyledComponent emotion-0"
>
  <p>
    1
    : 
    john
  </p>
  <p>
    2
    : 
    lisa
  </p>
</div>
`;
